### PR TITLE
Fix Docker Build

### LINKED ISSUE #51752: Docker machine broke
Dockerfile is broken because BSQL got bwoinked, also the BYOND version is way outdated.

[My fork](https://github.com/Hamcha/crstation) has a fixed version (along a docker-compose setup for having a SQL database running alongside it) but it relies on my own fork of [byond-docker](https://hub.docker.com/u/hamcha/byond-docker) since the `tgstation/byond` image has not been updated in a while, so I think it might be worth updating that image on the hub before making a PR or something.

Dunno how much you care about this, on the discord it seemed I was the only one who uses this setup.

### PR DESCRIPTION
**Pull request by Cyberboss**
Fixes #51752 

- BYOND dependency bumped to 513.1526
- Moved sounds and icons loaded at runtime to `sounds/runtime` and `icons/runtime` respectively
- Removed BSQL from docker build
- Added docker build test and publish actions

https://github.com/tgstation/byond-docker will now automatically build new BYOND releases at midnight

Most changes are just instrument file renames.

Sidenote: We really need to add a check for resources we're loading at runtime, look at how many sounds I had to fix.

### CODE REVIEW

**Comment by Cyberboss** [comment]
And some of these sounds straight don't exist lmao

**Comment by Cyberboss** [comment]
You may not like it, but this is what peak performance looks like
image

File: .github/workflows/docker_test.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+jobs:
+  docker_build:
+    runs-on: ubuntu-latest
+    name: Build Artifacts Docker Image
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Build Docker Image
+        run: docker build . -f Dockerfile
```
Comment (AnturK) [inline .github/workflows/docker_test.yml:11]: Will this actually fail the action if problem occurs ?

File: code/game/sound.dm [hunk 2]
```diff
@@ -177,7 +177,7 @@
 			if("goose")
 				soundin = pick('sound/creatures/goose1.ogg', 'sound/creatures/goose2.ogg', 'sound/creatures/goose3.ogg', 'sound/creatures/goose4.ogg')
 			if("warpspeed")
-				soundin = 'sound/effects/hyperspace_begin.ogg'
+				soundin = 'sound/runtime/hyperspace/hyperspace_begin.ogg'
```
Comment (AnturK) [inline code/game/sound.dm:180]: Should this be in runtime if it's just included.

File: tools/travis/run_server.sh [hunk 3]
```diff
@@ -2,13 +2,13 @@
...
 
 tools/deploy.sh travis_test
+rm travis_test/*.dll
 mkdir travis_test/config
 
 #test config
 cp tools/travis/travis_config.txt travis_test/config/config.txt
 
 cd travis_test
-ln -s $HOME/libmariadb/libmariadb.so libmariadb.so
```
Comment (AnturK) [inline tools/travis/run_server.sh:11]: What was this for ?

**Review by AnturK — CHANGES_REQUESTED** [review]

File: .github/workflows/docker_test.yml (hunk 1 above)
Comment (Cyberboss) [reply .github/workflows/docker_test.yml:11]: Yep, will have a non-zero exit code

File: code/game/sound.dm (hunk 2 above)
Comment (Cyberboss) [reply code/game/sound.dm:180]: It's accessed using double quotes by shuttles.

File: tools/travis/run_server.sh (hunk 3 above)
Comment (Cyberboss) [reply tools/travis/run_server.sh:11]: BSQL memes

**Comment by AnturK** [comment]
Yeehaw

**Pull request merged**